src/x86_parser.rs

```diff
@@ -263,3 +285,59 @@ pub fn populate_name_to_instruction_map<'instruction>(
         }
     }
 }
+
+fn get_cache_dir() -> anyhow::Result<PathBuf> {
+    // grab the home directory and build off of that
+    let mut x86_cache_path = home::home_dir().ok_or(anyhow!("Home directory not found"))?;
+
+    x86_cache_path.push(".cache");
+    x86_cache_path.push("asm-lsp");
+
+    // create the ~/.cache/asm-lsp directory if it's not already there
+    fs::create_dir_all(x86_cache_path.clone())?;
+
+    Ok(x86_cache_path)
+}
+
+fn get_x86_docs_web(x86_online_docs: &str) -> anyhow::Result<String> {
+    debug!(
+        "Fetching further documentation from the web -> {}...",
+        x86_online_docs
+    );
+    // grab the info from the web
+    let contents = reqwest::blocking::get(x86_online_docs)?.text()?;
+    Ok(contents)
+}
+
+fn get_x86_docs_cache(x86_cache_path: &PathBuf) -> Result<String, std::io::Error> {
+    debug!(
+        "Fetching further documentation from the cache -> {}...",
```

Comment:
Could we make this an `info!` message ? Also, could we make  the message a bit more precise? something like `Fetching html page containing further documentation, from the cache -> {}  ...`